backend/src/test/java/org/eclipse/tractusx/puris/backend/erpadapter/logic/service/ErpAdapterRequestClientTest.java

```diff
@@ -0,0 +1,167 @@
+        erpAdapterRequestClient.sendRequest(erpAdapterRequest);
+        RecordedRequest request = mockWebServer.takeRequest(2, TimeUnit.SECONDS);
+
+        // then
+        Assertions.assertThat(request.getMethod()).isEqualTo("POST");
+
+        Assertions.assertThat(request.getHeader(apiKey)).isEqualTo(apiSecret);
+        Assertions.assertThat(request.getHeader("Content-type")).contains("application/json");
+
+        var pairs = request.getPath().substring(2).split("&");
+        Map<String, String> parameters = Stream.of(pairs)
+            .map(string -> {
+                var keyValue = string.split("=");
+                return Map.entry(keyValue[0], keyValue[1]);
+            }).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        Assertions.assertThat(parameters.size()).isEqualTo(5);
+        Assertions.assertThat(parameters.get("bpnl")).isEqualTo(supplierPartnerBpnl);
+        Assertions.assertThat(parameters.get("request-type")).isEqualTo(requestType);
+        Assertions.assertThat(parameters.get("samm-version")).isEqualTo(sammVersion);
+        Assertions.assertThat(parameters.get("request-timestamp")).isEqualTo(String.valueOf(erpAdapterRequest.getRequestDate().getTime()));
+        Assertions.assertThat(parameters.get("request-id")).isEqualTo(uuid.toString());
+
+
+        try (InputStream stream = request.getBody().inputStream()) {
+            JsonNode requestBodyNode = objectMapper.readTree(new String(stream.readAllBytes()));
+            Assertions.assertThat(requestBodyNode.get("material").asText()).isEqualTo(matNbrCustomer);
+            Assertions.assertThat(requestBodyNode.get("direction").asText()).isEqualTo(DirectionCharacteristic.INBOUND.toString());
+            Assertions.assertThat(requestBodyNode.get("responseUrl").asText()).isEqualTo(erpResponseUrl);
+        }
+    }
+
+    @Test
+    public void test_with_faulty_credentials_should_fail() throws Exception {
+        // given
+        UUID uuid = UUID.randomUUID();
+        ErpAdapterRequest erpAdapterRequest = ErpAdapterRequest.builder()
+            .requestDate(new Date())
+            .partnerBpnl(supplierPartnerBpnl)
+            .id(uuid)
+            .directionCharacteristic(DirectionCharacteristic.INBOUND)
+            .ownMaterialNumber(matNbrCustomer)
+            .requestType(requestType)
+            .sammVersion(sammVersion)
+            .build();
+
+        // when
+        Mockito.when(variablesService.getErpAdapterUrl()).thenReturn(mockWebServer.url("/").toString());
+        Mockito.when(variablesService.getErpAdapterAuthKey()).thenReturn("wrong-key");
+        Mockito.when(variablesService.getErpAdapterAuthSecret()).thenReturn(apiSecret);
+        Mockito.when(variablesService.getErpResponseUrl()).thenReturn(erpResponseUrl);
+        erpAdapterRequestClient.sendRequest(erpAdapterRequest);
+        RecordedRequest request = mockWebServer.takeRequest(2, TimeUnit.SECONDS);
```

Comment:
Just a comment: Ok being honest I confused things. on client level the negative case makes less sense. In my head I was already thinking about the service level where we can assert how our service works with negative test case.

Reply:
Yeah, that testcase I added yesterday is not too useful ;)
I think it's best to remove it, since it takes quite some time (~ 10 sec) and it's probably not worth it

Reply:
I agree on that. Sorry for waisting your time :/